### PR TITLE
reference: improve documents about pessimistic transaction

### PR DESCRIPTION
**Pull request by youjiali1995**
### What is changed, added or deleted? 

Improve documents about pessimistic transaction.

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [ ] This PR is to align with:
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [ ] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

**Comment by youjiali1995** [comment]
@coocood @cfzjywxk PTAL

File: dev/reference/configuration/tidb-server/mysql-variables.md [hunk 1]
```diff
@@ -189,6 +189,7 @@ MySQL 系统变量 (System Variables) 是一些系统参数，用于调整数据
 | time_zone | GLOBAL \| SESSION | 数据库所使用的时区 |
 | tx_isolation | GLOBAL \| SESSION | 事务隔离级别 |
 | max\_execution\_time | GLOBAL \| SESSION | 语句超时时间，单位为毫秒 |
+| lock\_wait\_timeout | GLOBAL \| SESSION | 悲观事务语句等锁时间，单位为妙 |
```
Comment (cfzjywxk) [inline dev/reference/configuration/tidb-server/mysql-variables.md]: 单位为秒

File: dev/reference/configuration/tikv-server/configuration-file.md [hunk 2]
```diff
@@ -1072,3 +1072,21 @@ import 相关的配置项。
...
+## pessimistic-txn
+
+### `enabled`
+
++ 开启悲观事务支持，悲观事务使用方法请参考 [TiDB 悲观事务模式](/dev/reference/transactions/transaction-pessimistic.md)。
++ 默认值：true
+
+### `wait-for-lock-timeout`
+
++ 悲观事务在 TiKV 中等待其他事务释放锁的最长时间，单位为毫秒。
```
Comment (cfzjywxk) [inline dev/reference/configuration/tikv-server/configuration-file.md]: 这里是否需要写一下，超时之后行为和tidb的表现

**Comment by youjiali1995** [comment]
@cfzjywxk PTAL again.

File: dev/reference/configuration/tikv-server/configuration-file.md [hunk 3]
```diff
@@ -1072,3 +1072,21 @@ import 相关的配置项。
...
+## pessimistic-txn
+
+### `enabled`
+
++ 开启悲观事务支持，悲观事务使用方法请参考 [TiDB 悲观事务模式](/dev/reference/transactions/transaction-pessimistic.md)。
++ 默认值：true
+
+### `wait-for-lock-timeout`
+
++ 悲观事务在 TiKV 中等待其他事务释放锁的最长时间，单位为毫秒。若超时则会返回错误给 TiDB 并由 TiDB 重试加锁，语句最长等锁时间由 `lock_wait_timeout` 控制。
```
Comment (cfzjywxk) [inline dev/reference/configuration/tikv-server/configuration-file.md]: `innodb_lock_wait_timeout`

**Pull request merged**